controllers/productController.ts

```diff
@@ -14,28 +15,256 @@ const getAllProducts = async (req: Request, res: Response) => {
 
 // This method returns a specific product by ID
 const getProductById = async (req: Request, res: Response) => {
-    const productId=req.params.id;
+  const productId = req.params.id;
+
+  db.Product.findByPk(productId)
+    .then((product: any) => {
+      if (!product) {
+        res.status(404).json({ error: "product not found" });
+        return;
+      }
+      res.json({ product });
+    })
+    .catch((error: Error) => {
+      // tslint:disable-next-line:no-console
+      console.error("Error finding product:", error);
+      res.status(500).json({ error: "Internal server error" });
+    });
 };
 
 // This method returns all products for a specific category ID
 const getProductsByCategoryId = async (req: Request, res: Response) => {
-  return;
+  const categoryId = req.params.id;
+
+  db.Product.findAll({ where: { categoryId } })
+    .then((products: any) => {
+      if (!products || products.length === 0) {
+        res
+          .status(404)
+          .json({ error: "No products found for the given category ID" });
+        return;
+      }
+      res.json({ products });
+    })
+    .catch((error: Error) => {
+      // tslint:disable-next-line:no-console
+      console.error("Error finding products by category ID:", error);
+      res.status(500).json({ error: "Internal server error" });
+    });
 };
 
 // This method returns all products for a specific brand ID
 const getProductsByBrandId = async (req: Request, res: Response) => {
-  return;
+  const brandId = req.params.id;
+
+  db.Product.findAll({ where: { brandId } })
+    .then((products: any) => {
+      if (!products || products.length === 0) {
+        res
+          .status(404)
+          .json({ error: "No products found for the given brand ID" });
+        return;
+      }
+      res.json({ products });
+    })
+    .catch((error: Error) => {
+      // tslint:disable-next-line:no-console
+      console.error("Error finding products by brand ID:", error);
+      res.status(500).json({ error: "Internal server error" });
+    });
 };
 
 // This method creates a new product
 const createNewProduct = async (req: Request, res: Response) => {
   return;
 };
+const getNewArrivals = (req: Request, res: Response) => {
+  const threeMonthsAgo = new Date();
+  threeMonthsAgo.setMonth(threeMonthsAgo.getMonth() - 3);
 
+  db.Product.findAll({
+    where: {
+      createdAt: {
+        [Op.gt]: threeMonthsAgo,
+      },
+    },
+    attributes: [
```

Comment:
no need to add attributes if you return them all. However, I think we have to exclude some like updatedAt, you can modify it as:

attributes: { exclude: ['updatedAt'] },